File: python/src/player.py
from core.model import PlayerModel
from log import Log
from quests.example import example_quest, example_websocket_quest


class Player(PlayerModel):
    """
    Implementation of the Player model
    """

    def __init__(self):
        """
        Construct a Kodeventure Player
        """

        super().__init__()

    def load_quests(self):
        """
        Register all your quest handlers here
        """

        # Add a quest handler for a regular HTTP POST request
        self.add_quest('POST', '/my-simple-quest', example_quest)

        # Add a quest handler for a quest request over websocket
        self.ws.quest_handlers.append(example_websocket_quest)

        Log.info('Quests loaded')

File: javascript/src/player.js
const model = require('./core/model')
const example = require('./quests/example')


/**
 * Implementation of the Player model
 */
class Player extends model.PlayerModel {
  /**
   * Construct a Kodeventure Player
   */
  constructor() {
    super()
  }

  /**
   * Register all your quest handlers here
   */
  loadQuests() {
    // Add a quest handler for a regular HTTP POST request
    this.express.post('/my-simple-quest', example.exampleQuestHandler)

    // Uncomment to use a class based quest handler instead of the above
    // this.exampleQuest = new example.ExampleQuestHandler(this)

    // Add a quest handler for a quest request over websocket
    this.ws.questHandlers.push(example.exampleWsQuestHandler)
  }
}


module.exports = {
  Player: Player
}